### PR TITLE
[Main Course] - Kdd cup2021 Solution

### PR DESCRIPTION
**Pull request by jhqian0909**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Post for Kdd cup2021 Solution

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] This change requires a documentation update

# How Has This Been Tested?

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by Realvincentyuan — APPROVED** [review]
希望再补充一些细节

File: docs/Main_Course/Data_Analytics/2021-10-30-kdd-cup2021.md [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    作者：Void，发布于2021-10-30，阅读时间：约10分钟，微信公众号文章链接：[:fontawesome-solid-link:](https://mp.weixin.qq.com/s/XwtXpa1hOKrN6fIC-zpyKw)
+
+## 1 引言
+
+我们在[今天你异常了吗](https://mp.weixin.qq.com/s/XwtXpa1hOKrN6fIC-zpyKw)这篇文章中曾介绍过KDD Cup2021的时间序列异常检测竞赛。竞赛要求我们在250个时间序列中找出每个序列中的唯一异常点所在区间。  
+最近，赛事公布了最终高分团队的解决方案。扶我起来我还能学，今天就让我们一起来看一下这些高分方案吧。
+
+## 2 背景介绍
+
+第一名是来自深兰科技(DeepBlue Technology)的团队。这家公司有一些AI机器人的产品，也很热衷于参加各类竞赛。它们的解决方案是：TsaDetect: An efficient and flexible time series anomaly detection framework。
```
Comment (jz263) [inline docs/Main_Course/Data_Analytics/2021-10-30-kdd-cup2021.md:17]: 能否把深兰科技公众号的文章链接放在我们文章里作为reference: https://mp.weixin.qq.com/s/Z8rnDiw3_w67633EmsOzVg

**Review by jz263 — APPROVED** [review]

**Review by Tina0330 — APPROVED** [review]
LGTM，thanks！

File: docs/Main_Course/Data_Analytics/2021-10-30-kdd-cup2021.md (hunk 1 above)
Comment (jhqian0909) [reply docs/Main_Course/Data_Analytics/2021-10-30-kdd-cup2021.md:17]: 已加入

File: docs/Main_Course/Data_Analytics/2021-10-30-kdd-cup2021.md [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+---
+
+# KDD Cup2021时间序列异常检测竞赛总结
+
+!!! info 
+    作者：Void，发布于2021-10-30，阅读时间：约10分钟，微信公众号文章链接：[:fontawesome-solid-link:](https://mp.weixin.qq.com/s/XwtXpa1hOKrN6fIC-zpyKw)
+
+## 1 引言
+
+我们在[今天你异常了吗](https://mp.weixin.qq.com/s/XwtXpa1hOKrN6fIC-zpyKw)这篇文章中曾介绍过KDD Cup2021的时间序列异常检测竞赛。竞赛要求我们在250个时间序列中找出每个序列中的唯一异常点所在区间。
```
[the remark this answers is not in the document]
Comment (jhqian0909) [reply docs/Main_Course/Data_Analytics/2021-10-30-kdd-cup2021.md:12]: 好的

File: docs/Main_Course/Data_Analytics/2021-10-30-kdd-cup2021.md [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+  <img src="https://cdn.jsdelivr.net/gh/BulletTech2021/Pics/img/6.png" width="500" />
+</figure>
+
+### 4.2 Matrix Profile算法
+
+Matrix Profile算法在上篇中已经详细地介绍过，具体的思想是计算每个区间和其他区间的欧式距离，得到异常区间。
+
+### 4.3 基于回归的算法
+
+基于回归的算法是指用模型做一个回归问题，来预测某个点的值。然后在测试集用预测值和真实值做差，得到残差。高分方案尝试了不同的算法，最终选择的是GBDT和CNN算法。值得一提的是，X可以取预测值左右两边的数据。同时异常点附近可能都是异常的，为此，X和预测值之间留了一点gap。
```
[the remark this answers is not in the document]
Comment (jhqian0909) [reply docs/Main_Course/Data_Analytics/2021-10-30-kdd-cup2021.md:89]: 残差最大的点就是异常点。
高分方案是参考的这篇https://mp.weixin.qq.com/s/Jf7ckFpP7pqubOBN7OkgTQ
并没有找到更详细的内容了，我再去找找吧。

**Pull request merged**